decidim-core/lib/decidim/form_builder.rb

```diff
@@ -131,29 +131,43 @@ def categories_select(name, collection, options = {})
       select(name, @template.options_for_select(categories, selected: selected, disabled: disabled), options, html_options)
     end
 
-    # Public: Generates a select field with the scopes.
+    # Public: Generates a picker field for scope selection.
     #
-    # name       - The name of the field (usually scope_id)
-    # collection - A collection of scopes.
-    # options    - An optional Hash with options:
-    # - prompt   - An optional String with the text to display as prompt.
+    # name          - The name of the field (usually scope_id)
+    # options       - An optional Hash with options:
+    # - multiple    - Multiple mode, to allow multiple scopes selection.
+    # - label       - Show label?
+    #
+    # Also it should receive a block that returns a Hash with :url and :text for each selected scope (and for null scope for prompt)
     #
     # Returns a String.
-    def scopes_select(name, options = {})
-      selected = object.send(name)
-      if selected.present?
-        selected = selected.values if selected.is_a?(Hash)
-        selected = [selected] unless selected.is_a?(Array)
-        scopes = Decidim::Scope.where(id: selected.map(&:to_i)).map { |scope| [scope.name[I18n.locale.to_s], scope.id] }
-      else
-        scopes = []
-      end
-      prompt = options.delete(:prompt)
-      remote_path = options.delete(:remote_path) || false
-      multiple = options.delete(:multiple) || false
-      html_options = { multiple: multiple, class: "select2", "data-remote-path" => remote_path, "data-placeholder" => prompt }
+    def scopes_picker(attribute, options = {})
```

Comment:
This is fine like this - but when the complexity of these things arises and is mostly markup, what I usually do is extract the code into a partial somewhere and just render that. This way you can leverage the clearness of the markup in a partial and have the logic be passed via locals (and prepopulated before).

It's something to consider the next time we step in this code 👍